### PR TITLE
Add method to return TimeSeriesContainer from a metricCollectionTimeSeries

### PR DESCRIPTION
**Pull request by sevans40**
convert methods to static throughout DssDatabase
add method to return TimeSeriesContainer from metricCollectionTimeSeries in order to perform timeseries math on the mcts. Also  refactor the write method to include getTimeSeriesContainer to avoid code duplication.

### CODE REVIEW

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java [hunk 1]
```diff
@@ -439,7 +441,33 @@ public void write(hec.metrics.MetricCollectionTimeSeries[] metricsArray) throws
...
 
+            for (int i = 0; i < statsAsSeparateTimeseries.length; i++) {
+                TimeSeriesContainer tsc = getTimeSeriesContainer(mcts.getTimeSeriesIdentifier(), mc, statsAsSeparateTimeseries[i], mc.getValues()[i]);
+                tscList.add(tsc);
+            }
+        }
+        return tscList.toArray(new TimeSeriesContainer[0]);
+    }
+
+    private static TimeSeriesContainer getTimeSeriesContainer(RecordIdentifier identifier, MetricCollection mc, String stat, float[] values) {
```
Comment (Brennan1994) [inline ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java:458]: Is this getting a timeseries container from a DSSDatabase, or creating a timeseries container? Might rethink the name. Maybe this belongs as a constructor on TimeSeriesContainer?

**Review by Brennan1994 — CHANGES_REQUESTED** [review]

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java (hunk 1 above)
Comment (ktarbet) [reply ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java:458]: This is converting MetricCollectionTimeSeries to TimeSeriesContainer[] 

How about the name convertToTimeSeriesContainer  ?

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java [hunk 2]
```diff
@@ -439,7 +441,33 @@ public void write(hec.metrics.MetricCollectionTimeSeries[] metricsArray) throws
         CatalogIsUpToDate = false;
     }
 
+    public static TimeSeriesContainer[] toTimeSeriesContainer(MetricCollectionTimeSeries mcts) {
```
[the remark this answers is not in the document]
Comment (Brennan1994) [reply ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java:442]: Understood. I think that's a good call. I'd maybe leave myself a comment on the method to remind me that's why it's here if I have to go back, but either way we can consider comment this resolved.

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java (hunk 1 above)
Comment (Brennan1994) [reply ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java]: I approve. That seems like a good, descriptive name.

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sevans40) [reply ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java:447]: comment to method added.

**Review by ktarbet — APPROVED** [review]

**Review by ktarbet — APPROVED** [review]

**Review by Brennan1994 — APPROVED** [review]

**Pull request merged**